iothub/service/tests/Configurations/ConfigurationsClientTests.cs

```diff
@@ -0,0 +1,608 @@
+
+            // act
+            Func<Task> act = async () => await configurationsClient.CreateAsync(null).ConfigureAwait(false);
+
+            // assert
+            await act.Should().ThrowAsync<ArgumentNullException>().ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task ConfigurationClients_GetAsync()
+        {
+            //arrange
+            string configurationId = Guid.NewGuid().ToString().ToLower(); // Configuration Id characters must be all lower-case.
+            var configuration = new Configuration(configurationId)
+            {
+                Priority = 2,
+                Labels = { { "testLabelName", "testLabelValue " } },
+                TargetCondition = "deviceId = testId"
+            };
+
+            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();
+            mockCredentialProvider
+                .Setup(getCredential => getCredential.GetAuthorizationHeader())
+                .Returns(s_validMockAuthenticationHeaderValue);
+            var mockHttpRequestFactory = new HttpRequestMessageFactory(s_httpUri, "");
+            using var mockHttpResponse = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = HttpMessageHelper.SerializePayload(configuration)
+            };
+            var mockHttpClient = new Mock<HttpClient>();
+            mockHttpClient
+                .Setup(restOp => restOp.SendAsync(It.IsAny<HttpRequestMessage>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(mockHttpResponse);
+
+            var configurationsClient = new ConfigurationsClient(
+                HostName,
+                mockCredentialProvider.Object,
+                mockHttpClient.Object,
+                mockHttpRequestFactory,
+                s_retryHandler);
+
+            // act
+            Func<Task> act = async () => await configurationsClient.GetAsync(configurationId).ConfigureAwait(false);
+
+            // assert
+            await act.Should().NotThrowAsync().ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task ConfigurationClients_GetAsync_NullConfigurationIdThrows()
+        {
+            // arrange
+            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();
+            var mockHttpRequestFactory = new Mock<HttpRequestMessageFactory>();
+            var mockHttpClient = new Mock<HttpClient>();
+
+            var configurationsClient = new ConfigurationsClient(
+                HostName,
+                mockCredentialProvider.Object,
+                mockHttpClient.Object,
+                mockHttpRequestFactory.Object,
+                s_retryHandler);
+
+            // act
+            Func<Task> act = async () => await configurationsClient.GetAsync(null).ConfigureAwait(false);
+
+            // assert
+            await act.Should().ThrowAsync<ArgumentNullException>().ConfigureAwait(false);
```

Comment:
you can also check the expected exception thrown here with a particular parameter name. See sample [here](https://fluentassertions.com/exceptions/) for calling with `.WithParameterName()`.

Reply:
here and below.